src/Components/CreateImageWizard/validators.ts

```diff
@@ -71,6 +71,19 @@ export const isUserNameValid = (userName: string) => {
   return isLengthValid && isNotNumericOnly && isPatternValid;
 };
 
+export const isUserGroupValid = (group: string) => {
+  // see `man groupadd` for the exact specification
+  if (!group) {
```

Comment:
I know we may use this pattern elsewhere but I feel like it is an antipattern.

It is tempting to use this because `''` is the empty state and so we don't want to make it invalid. But I think this creates a tight coupling. The fact that `''` is the empty state is an implementation detail - the `isWhateverValid` validator functions shouldn't know about that.

Instead, I propose that we check at the call site whether or not the field is empty or full and then call the validator if appropriate.

Reply:
Great point, removed.

Reply:
Opened a follow up to remove the pattern in the rest of the validators https://github.com/osbuild/image-builder-frontend/pull/3031